op-service/txmgr/txmgr.go

```diff
@@ -819,6 +819,12 @@ func (m *SimpleTxManager) increaseGasPrice(ctx context.Context, tx *types.Transa
 			"gasFeeCap", bumpedFee, "gasTipCap", bumpedTip)
 	}
 
+	if tx.Gas() > gas {
+		// Don't bump the gas limit down if the passed-in gas limit is higher than
+		// what was originally specified.
+		gas = tx.Gas()
+	}
+
```

Comment:
The current behavior is that we take the gas estimation and use it when recreating a new transaction from the old. New behavior is that it uses the larger of [newEstimation, oldGas]. Effectively this means gas can't go down between iterations. Given that the function is called `increaseGasPrice`, seems like the right change.

I am unaware of any current use for Transaction Manager that would break with this change. I *do* think it might create gas inefficiencies in volatile markets, as a single large estimation could persist into cheaper territory. I think that behavior is potentially good, as it means the transaction is more competitive for longer, and will likely get included quickly. Regardless, this is a behavior modification that **might** be good to put into the [SimpleTxMgr Config](https://github.com/ethereum-optimism/optimism/blob/9452aa66dc7a2e0deb63bca4bf5c7cfc35c3211c/op-service/txmgr/cli.go#L365). That way, callers may decide to use either behavior if they are cost-sensitive.

And there's a suite of tests in `txmgr_test` that use `doGasPriceIncrease` that you could extend to show that the behavior is as expected, whether or not you choose to go with a configurable option. You'd need to extend `doGasPriceIncrease` to optionally modify the transaction used in the test.

Reply:
Awesome thank you!